iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/SV.java

```diff
@@ -84,132 +74,341 @@ public double getA() {
-        Complex v2 = v1p.subtract(z.multiply(i2p)); // v2p=v1p-z*i2
-        Complex i2 = i2p.subtract(y2.multiply(v2)); // i2=i2p-y2*v2
-        return v2.multiply(3).multiply(i2.conjugate()); // s2=3*v2*conj(i2)
+    public SV otherSide(double r, double x, double g1, double b1, double g2, double b2, double rho) {
+        return otherSide(r, x, g1, b1, g2, b2, rho, 0.0);
     }
 
-    private Complex computeS2(Complex y, Complex z, double rho) {
-        return computeS2(y, new Complex(0, 0), z, rho);
+    public SV otherSide(double r, double x, double g1, double b1, double g2, double b2, double rho, double alpha) {
+        LinkData.BranchAdmittanceMatrix adm = LinkData.calculateBranchAdmittance(r, x, 1 / rho, -alpha, 1.0, 0.0,
+            new Complex(g1, b1), new Complex(g2, b2));
+        return otherSide(adm);
     }
 
-    public SV otherSide(double r, double x, double g, double b, double rho) {
-        Complex z = new Complex(r, x); // z=r+jx
-        Complex y = new Complex(g, b); // y=g+jb
-        Complex u2 = computeU2(y, z, rho);
-        Complex s2 = computeS2(y, z, rho);
-        return new SV(-s2.getReal(), -s2.getImaginary(), u2.abs(), Math.toDegrees(u2.getArgument()));
+    // Has been modified
+    // Before: otherSide(l.getR(), l.getX(), l.getG1() + l.getG2(), l.getB1() + l.getB2(), 0.0, 0.0, 1.0, 0.0);
+    public SV otherSide(Line l) {
+        return otherSide(l.getR(), l.getX(), l.getG1(), l.getB1(), l.getG2(), l.getB2(), 1.0, 0.0);
     }
 
-    public SV otherSide(double r, double x, double g1, double b1, double g2, double b2, double rho) {
-        Complex z = new Complex(r, x); // z=r+jx
-        Complex y1 = new Complex(g1, b1); // y1=g1+jb1
-        Complex y2 = new Complex(g2, b2); // y2=g2+jb2
-        Complex u2 = computeU2(y1, y2, z, rho);
-        Complex s2 = computeS2(y1, y2, z, rho);
-        return new SV(-s2.getReal(), -s2.getImaginary(), u2.abs(), Math.toDegrees(u2.getArgument()));
+    /**
+     * @deprecated Not used anymore. Use
+     * {@link SV#otherSide(Line)}
+     */
+    @Deprecated
+    public SV otherSideY1Y2(Line l) {
+        throw new PowsyblException("Deprecated. Not used anymore");
     }
 
     public SV otherSide(TwoWindingsTransformer twt) {
-        return otherSide(getR(twt), getX(twt), getG(twt), getB(twt), getRho(twt));
+        return otherSide(getR(twt), getX(twt), getG(twt), getB(twt), 0.0, 0.0, getRho(twt), getAlpha(twt));
     }
 
-    public SV otherSide(Line l) {
-        return otherSide(l.getR(), l.getX(), l.getG1() + l.getG2(), l.getB1() + l.getB2(), 1);
+    public SV otherSide(TwoWindingsTransformer twt, boolean splitShuntAdmittance) {
+        if (splitShuntAdmittance) {
+            return otherSide(getR(twt), getX(twt), getG(twt) * 0.5, getB(twt) * 0.5, getG(twt) * 0.5, getB(twt) * 0.5, getRho(twt), getAlpha(twt));
+        } else {
+            return otherSide(getR(twt), getX(twt), getG(twt), getB(twt), 0.0, 0.0, getRho(twt), getAlpha(twt));
```

Comment:
Maybe rewrite else as:

`... else return otherSide(twt)`

to make clear that splitShuntAdmittance is not the "default" ?

Reply:
Done